.github/workflows/client-celo.yml

```diff
@@ -51,14 +50,29 @@ jobs:
             go-build-env-celo \
             gotestsum -- -tags=celo,musl ./...
 
-      # This step is executed after the tests as we want to configure it eventually
-      # as image publication step.
-      - name: Build Docker Runtime Image
+      - name: Login to Google Container Registry
+        if:  |
+          startsWith(github.ref, 'refs/heads/releases/alfajores')
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.GCR_REGISTRY_URL }}
+          username: _json_key
+          password: ${{ secrets.KEEP_TEST_GCR_JSON_KEY }}
+
+      - name: Build and publish Docker Runtime Image
         uses: docker/build-push-action@v2
+        env:
+          IMAGE_NAME: 'keep-ecdsa-celo'
+          GOOGLE_PROJECT_ID: ${{ secrets.KEEP_TEST_GOOGLE_PROJECT_ID }}
         with:
-          tags: keep-ecdsa-celo
-          labels: |
-            revision=${{ github.sha }}
-          # TODO: Check branch name and publish to a registry accordingly to the
-          # environment.
-          # push: true # publish to registry
\ No newline at end of file
+          # GCR image should be named according to following convention:
+          # HOSTNAME/PROJECT-ID/IMAGE:TAG
+          # We don't use TAG yet, will be added at later stages of work on RFC-18.
+          tags: ${{ secrets.GCR_REGISTRY_URL }}/${{ env.GOOGLE_PROJECT_ID }}/${{ env.IMAGE_NAME }}
+          labels: revision=${{ github.sha }}
+          build-args: |
+            HOST_CHAIN=celo
+            REVISION=${{ github.sha }}
+          # VERSION= ? TODO: Configure version, sample: 1.7.6
+          push: |
+            ${{ startsWith(github.ref, 'refs/heads/releases/alfajores') }}
```

Comment:
We could simplify branch name check with: 
```
${{ startsWith(github.head_ref, 'releases/alfajores') }}
```

But it's something that I'd not block this PR with.

cc: @michalinacienciala